### PR TITLE
React questions index

### PR DESCRIPTION
**Pull request by logtheman**
Added package for precompiling scss into css and added Bootstrap through NPM. Still need to work on getting the bootstrap JS files incorporated. Might need a JS loader. 

Will build the navbar back in with React next.

Please cd client and run NPM install to see changes.

### CODE REVIEW

File: client/src/components/common/PageBanner.jsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import logo from '../../images/carrot_2.png';
+
+const PageBanner = (props) => {
```
Comment (muffy) [inline client/src/components/common/PageBanner.jsx]: Our (Instacart) default for indentation for all languages is 2 spaces - adjust your editor and reformat.

File: client/src/components/questions/QuestionIndexRow.jsx [hunk 2]
```diff
@@ -1,14 +1,13 @@
 import React from 'react';
 
 const QuestionIndexRow = (props) => {
-	console.log("props from index: ", props);
 	return (
-		<div id="question" className="card d-block img-fluid mb-2">
```
Comment (muffy) [inline client/src/components/questions/QuestionIndexRow.jsx]: Ids should be unique - I see you changed this anyway, but if you want an id, classically you simply use the row id from the db, combined with some string, i.e. `question_123`

File: client/src/stylesheets/questions.scss [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+  		flex: 8;
+  		font-family: 'Muli', sans-serif;
+  	  font-weight: 300;
+  		color: white;
+  	  text-shadow: 1.5px 1.5px 0 black; 
+  	  margin-bottom: 0px;
+  	  font-size: $banner-text-sm;
+  }
+
+  @media(min-width: 768px){
```
Comment (muffy) [inline client/src/stylesheets/questions.scss]: Nice, have you actually tried it out on different screen sizes?

File: client/src/components/common/PageBanner.jsx (hunk 1 above)
Comment (logtheman) [inline client/src/components/common/PageBanner.jsx]: Actually don't know why it looks so strange here, but my editor is set to two spaces per tab and looks very different in my editor.

image

File: client/src/stylesheets/questions.scss (hunk 3 above)
Comment (logtheman) [inline client/src/stylesheets/questions.scss]: Oh yeah, I've been trying on multiple screen sizes and been using chrome developer tools to make sure break points make sense and cells phones would look okay. Still some work to be done there, but the groundwork is good.

**Comment by jerflan** [comment]
Yay no errors! However, @muffy, maybe you or @wrfo (Will, who is not part of this repo apparently), has some good info regarding proper set for working with react on rails. The last few merges have been fairly precarious, and I think it's because I'm not quite sure what's going on.

No issues on my end other than that.

**Comment by logtheman** [comment]
@muffy, @jerflan - any additional comments or concerns? If not, I can clear the package.json conflict and merge.

**Comment by jerflan** [comment]
I think we should make sure it works with Heroku first, no? Unless you have reason to believe that something in this merge would fix that.

**Pull request merged**